### PR TITLE
3987 group bulk export does not apply  typefilter searches to members of the group

### PR DESCRIPTION
**Pull request by StevenXLi**
- Added IT for this issue
- changed the method to get all members from a group to account for the typefilter
- changed unit tests to reflect the code change

### CODE REVIEW

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/bulk/export/svc/JpaBulkExportProcessor.java [hunk 1]
```diff
@@ -150,8 +153,7 @@ public Iterator<ResourcePersistentId> getResourcePidIterator(ExportPIDIteratorPa
 					pids.add(resultIterator.next());
 				}
 			}
-		}
-		else if (theParams.getExportStyle() == BulkDataExportOptions.ExportStyle.GROUP) {
+		} else if (theParams.getExportStyle() == BulkDataExportOptions.ExportStyle.GROUP) {
```
Comment (tadgh) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/bulk/export/svc/JpaBulkExportProcessor.java:156]: thanks for fixing this bracket 👍🏻

File: hapi-fhir-jpaserver-test-r4/src/test/java/ca/uhn/fhir/jpa/bulk/BulkDataExportTest.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+			for (String binaryId : binaryIds) {
+				Binary binary = myBinaryDao.read(new IdType(binaryId));
+				assertEquals(Constants.CT_FHIR_NDJSON, binary.getContentType());
+				String contents = new String(binary.getContent(), Constants.CHARSET_UTF8);
+				ourLog.info("Next contents for type {} :\n{}", binary.getResourceType(), contents);
+				assertThat(contents, Matchers.containsString("\"PF\""));
+				assertThat(contents, not(Matchers.containsString("\"PM\"")));
+			}
+		}
+	}
```
Comment (tadgh) [inline hapi-fhir-jpaserver-test-r4/src/test/java/ca/uhn/fhir/jpa/bulk/BulkDataExportTest.java:100]: very nice test. Can you also test the scenario where the type filter is omitted, that both patients are included? Also, ensure that a patient exists that is _not_ in the group, that doesnt get included. e.g. 

Create patient 1,2,3
Create group G, contains Patients 1,2

submit a request to export group G, ensure only patients 1/2 in there.

**Review by tadgh — APPROVED** [review]

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/bulk/export/svc/JpaBulkExportProcessor.java (hunk 1 above)
Comment (StevenXLi) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/bulk/export/svc/JpaBulkExportProcessor.java:156]: thank the formatter in intellij 😉

File: hapi-fhir-jpaserver-test-r4/src/test/java/ca/uhn/fhir/jpa/bulk/BulkDataExportTest.java (hunk 2 above)
Comment (StevenXLi) [reply hapi-fhir-jpaserver-test-r4/src/test/java/ca/uhn/fhir/jpa/bulk/BulkDataExportTest.java:134]: added new test

**Pull request merged**